frb_codegen/src/parser/mod.rs

```diff
@@ -343,6 +377,53 @@ fn item_method_to_function(
     if let Type::Path(p) = item_impl.self_ty.as_ref() {
         let struct_name = p.path.segments.first().unwrap().ident.to_string();
         let span = item_method.sig.ident.span();
+
+        // get/check inputs for mutability first
+        let mut is_mut = false;
+        // let inputs: syn::punctuated::Punctuated<_, _> = item_method
+        let inputs: Punctuated<FnArg, Comma> = item_method
+            .sig
+            .inputs
+            .iter()
+            .enumerate()
+            .map(|(i, input)| -> ParserResult<_> {
+                let mut segments = Punctuated::new();
+                segments.push(PathSegment {
+                    ident: Ident::new(struct_name.as_str(), span),
+                    arguments: PathArguments::None,
+                });
+                if let FnArg::Receiver(Receiver { mutability, .. }) = input {
+                    if i == 0 && mutability.is_some() {
+                        is_mut = true;
+                    }
+                    Ok(FnArg::Typed(PatType {
+                        attrs: vec![],
+                        pat: Box::new(Pat::Ident(PatIdent {
+                            attrs: vec![],
+                            by_ref: Some(syn::token::Ref { span }),
+                            mutability: *mutability,
+                            ident: Ident::new("that", span),
+                            subpat: None,
+                        })),
+                        colon_token: Colon { spans: [span] },
+                        ty: Box::new(Type::Path(TypePath {
+                            qself: None,
+                            path: Path {
+                                leading_colon: None,
+                                segments,
+                            },
+                        })),
+                    }))
+                } else {
+                    Ok(input.clone())
+                }
+            })
+            .collect::<ParserResult<Punctuated<_, _>>>()?;
+
+        if is_mut {
+            return Ok(None); // if the method is mutable, don't panic but skip it
```

Comment:
nit: maybe `log::warn` for it

Reply:
Refined in [3378af6](https://github.com/fzyzcjy/flutter_rust_bridge/pull/1145/commits/3378af69fd619676d405c259b9466efde3a56f88), which output log like this:
```
Mutable method detected: `SharedComplexEnumInAllBlocks::test_mutable_enum_method`, which won't be generated
Mutable method detected: `SharedComplexEnumInAllBlocks::test_mutable_ref_enum_method`, which won't be generated
Mutable method detected: `SharedStructInBlock2And3::test_mutable_method`, which won't be generated
Mutable method detected: `SharedStructInBlock2And3::test_mutable_ref_method`, which won't be generated
```